x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/text_classification/lang_ident_inference.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { InferenceBase } from '../inference_base';
+import { processResponse } from './common';
+import { getGeneralInputComponent } from '../text_input';
+import { getLangIdentOutputComponent } from './lang_ident_output';
+import type { TextClassificationResponse, RawTextClassificationResponse } from './common';
+
+export class LangIdentInference extends InferenceBase<TextClassificationResponse> {
+  private numTopClasses = this.model.inference_config?.classification?.num_top_classes || 5;
+
+  public async infer() {
+    try {
+      this.setRunning();
+      const inputText = this.inputText$.value;
```

Comment:
You should not use `value` with observables. `BehaviourSubject` has `.getValue` method, you can try it out. But ideally, you should take another approach with streams, that involves the `withLatestFrom` operator.

Reply:
I've updated this to use `getValue`
This file has only moved location in this PR and so I'd rather not rewrite how the inference class works in this PR.
I can look at follow up work to improve how this works with observables.